xarray/core/dataset.py

```diff
@@ -306,7 +309,7 @@ def load(self):
         """
         # access .data to coerce everything to numpy or dask arrays
         all_data = dict((k, v.data) for k, v in self.variables.items())
-        lazy_data = dict((k, v) for k, v in all_data.items()
+        lazy_data = OrderedDict((k, v) for k, v in all_data.items()
```

Comment:
on line 318, you're iterating on the values of lazy_data to generate the evaluated_data list, and then you're recombining the two by iterating on the keys of lazy_data on line 320.
_Typically_, if you iterate on a dict twice in a row you _should_ get the same order, but there's nothing guaranteeing that. So unless you use a container with order you're effectively relying on an implementation detail of CPython.